test/tests/parallel/set1/spam_virus/rspamd_full.bats

```diff
@@ -108,34 +109,41 @@ function teardown_file() { _default_teardown ; }
 }
 
 @test 'normal mail passes fine' {
-  _service_log_should_contain_string 'rspamd' 'F \(no action\)'
+  _service_log_should_contain_string 'rspamd' 'F (no action)'
 
-  _print_mail_log_for_id "${MAIL_ID_PASS}"
+  _print_mail_log_for_msgid 'rspamd-test-email-pass'
   assert_output --partial "stored mail into mailbox 'INBOX'"
 
   _count_files_in_directory_in_container /var/mail/localhost.localdomain/user1/new/ 1
 }
 
 @test 'detects and rejects spam' {
-  _service_log_should_contain_string 'rspamd' 'S \(reject\)'
+  _service_log_should_contain_string 'rspamd' 'S (reject)'
   _service_log_should_contain_string 'rspamd' 'reject "Gtube pattern"'
 
-  _print_mail_log_for_id "${MAIL_ID_SPAM}"
+  _print_mail_log_of_queue_id_from_msgid 'dms-test-email-spam'
   assert_output --partial 'milter-reject'
   assert_output --partial '5.7.1 Gtube pattern'
+  refute_output --partial "stored mail into mailbox 'INBOX'"
+
+  _print_mail_log_for_msgid 'dms-test-email-spam'
+  assert_failure
 
   _count_files_in_directory_in_container /var/mail/localhost.localdomain/user1/new/ 1
 }
 
 @test 'detects and rejects virus' {
-  _service_log_should_contain_string 'rspamd' 'T \(reject\)'
+  _service_log_should_contain_string 'rspamd' 'T (reject)'
   _service_log_should_contain_string 'rspamd' 'reject "ClamAV FOUND VIRUS "Eicar-Signature"'
 
-  _print_mail_log_for_id "${MAIL_ID_VIRUS}"
+  _print_mail_log_of_queue_id_from_msgid 'rspamd-test-email-virus'
   assert_output --partial 'milter-reject'
   assert_output --partial '5.7.1 ClamAV FOUND VIRUS "Eicar-Signature"'
   refute_output --partial "stored mail into mailbox 'INBOX'"
 
+  _print_mail_log_for_msgid 'dms-test-email-spam'
+  assert_failure
```

Comment:
While not exactly necessary, if the `assert_failure` did... fail, it would be good to catch this expected refute for context rather than a less helpful status failure. It also adds context on the log/action we're expecting to fail and test against.

```suggestion
  _print_mail_log_for_msgid 'dms-test-email-spam'
  refute_output --partial "stored mail into mailbox 'INBOX'"
  assert_failure
```